### PR TITLE
Branch dg update

### PR DESCRIPTION
**Pull request by yunpeng1234**
image

image

image

image

Might be the same as @Chang-CH pr but just to compare see if they are right

### CODE REVIEW

**Comment by yunpeng1234** [comment]
Guys help check the UML diagram thanks

**Comment by Jeromesyl** [comment]
For Model, there should be a VersionedInternWatcher that inherits from InternWatcher. VersionedInternWatcher actually replaces InternWatcher in ModelManager, so I think that the associations from ModelManager and UniquePersonList should be connected to VersionedInternWatcher instead?

**Comment by Chang-CH** [comment]
There are still traces of `person`. I already have edited PUML's on my PR.

File: docs/diagrams/ModelClassDiagram.puml [hunk 1]
```diff
@@ -5,50 +5,62 @@ skinparam arrowColor MODEL_COLOR
...
-Class ReadOnlyAddressBook
+Class InternWatcher
+Class ReadOnlyInternWatcher
 Class Model
 Class ModelManager
 Class UserPrefs
 Class ReadOnlyUserPrefs
-
-
 Class UniquePersonList
```
Comment (Chang-CH) [inline docs/diagrams/ModelClassDiagram.puml:17]: `Person` here

File: docs/diagrams/UiClassDiagram.puml [hunk 2]
```diff
@@ -32,6 +33,7 @@ UiManager .left.|> Ui
 UiManager -down-> "1" MainWindow
 MainWindow *-down->  "1" CommandBox
 MainWindow *-down-> "1" ResultDisplay
+MainWindow *-down-> "1" PersonDetailPanel
 MainWindow *-down-> "1" PersonListPanel
```
Comment (Chang-CH) [inline docs/diagrams/UiClassDiagram.puml:37]: `Person` here

**Review by Chang-CH — CHANGES_REQUESTED** [review]
Unless my PUML messed up as well these should be duplicate changes.

File: docs/diagrams/UiClassDiagram.puml (hunk 2 above)
Comment (yunpeng1234) [reply docs/diagrams/UiClassDiagram.puml:37]: Thats our ui component no?

**Comment by yunpeng1234** [comment]
New Diagrams

*DeleteSeq*
image

*DeleteAllSeq*
image

*EditAllSeq*
image

*ModelClass*
image

**Review by Chang-CH — APPROVED** [review]
LGTM

**Review by Chang-CH — APPROVED** [review]
LGTM

**Pull request merged**